publish.rb

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env ruby
+#
+# publish.rb
+#
+# This script publishes the ev3dev website to the gh-pages branch of your
+# personal fork. This allows you to share a preview of your changes with others.
+#
+# Usage: ./publish.rb <gh-user-name>
+
+require 'tmpdir'
+
+if ARGV.count != 1
+    STDERR.puts "Usage: ./publish.rb <gh-user-name>"
+    exit(1)
+end
+
+Dir.mktmpdir do |tmp|
+    FileUtils.cp_r "_site/.", tmp
+    Dir.chdir tmp
+    system "git init"
+    git_url = "git@github.com:#{ARGV[0]}/ev3dev.github.io.git"
+    system "git remote add origin #{git_url}"
```

Comment:
This creates a new local git repository in a temporary directory, so it can't conflict with anything.

I always use ssh for git, even on Windows. I don't like using https because you have to enter your password every time. I would be ok adding a command line switch so that you can choose one or the other.